File: sol/solution/0001-0100/0012. Integer to Roman/Solution.rs
impl Solution {
    pub fn int_to_roman(mut num: i32) -> String {
        let values = [(1000, "M"), (900, "CM"), (500, "D"), (400, "CD"), (100, "C"), (90, "XC"), (50, "L"), (40, "XL"), (10, "X"), (9, "IX"), (5, "V"), (4, "IV"), (1, "I")];
        let mut result = String::new();
        for &(val, sym) in values.iter() {
            while num >= val {
                result.push_str(sym);
                num -= val;
            }
        }
        result
    }
}

File: sol/solution/0001-0100/0012. Integer to Roman/Solution.swift
class Solution {
    func intToRoman(_ num: Int) -> String {
        let values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        let symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
        var num = num
        var result = ""
        for (i, value) in values.enumerated() {
            while num >= value {
                result += symbols[i]
                num -= value
            }
        }
        return result
    }
}